### ISSUE TITLE
False positive for "Unnecessary isinstance call"

### ISSUE THREAD

**Issue by mtomassoli** [microsoft/pyright#9272]
From version 1.1.380 onwards, Pyright erroneously reports the `isinstance` below as unnecessary:

```python
class A:
    ...
    
class B(A):
    ...

class C(A):
    ...

def f(elem: B | C, _type: type[A]) -> bool:
    return isinstance(elem, _type)

assert f(B(), B)
assert not f(B(), C)
```

That would be correct only if `type` were invariant.

**Comment by erictraut** [comment]
Thanks for the bug report and the clear code sample.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386
